LiskHQ/lisk-hub
# Implement add to bookmark page - Closes #2056

### What issue have I solved?

#2056

### How have I implemented/fixed it?

- Created new route `/bookmarks/add-bookmark`;
- Created form for adding a new bookmark;
- Validation of address is token based;
- Avatar is only generated for valid addresses, and is token based;
- Should show error if address is invalid or already bookmarked;
- Label only have validation of length, must be <= 20 character;
- Was decided with the design team, that we don't need the spinner and checkmark on the inputs on this page;
- If token is LSK and address belongs to a delegate, username is fetched and label field gets updated and becomes readonly;
- Switching token should clear the fields;

Obs: After successfully adding a new bookmark, it gets redirected to`/dashboard` page, due to the `/bookmarks` page being implemented in another PR: #2187 

### How has this been tested?

- Go to `/bookmarks/add-bookmark`;
- Fill the inputs with any desired data, it should have  the validations on both fields;
  - If address belongs to a delegate, label should be auto filled and readonly;
- After adding the bookmark, go to `/dashboard` to check  that the bookmark was added, since we don't have the bookmarks page yet;

### Review checklist
- The PR follows our [Test guide](/LiskHQ/lisk-hub/blob/development/docs/TEST_GUIDE.md)
- The PR follows our [CSS guide](/LiskHQ/lisk-hub/blob/development/docs/CSS_GUIDE.md)
